utils/user_management.py

```diff
@@ -0,0 +1,158 @@
+        '--password',
+        help=('Password for the username. If not specified password will '
+              'be generated'))
+    return parser
+
+
+def check_args(args):
+    if (args.username is None) or (args.pwdfile is None):
+        return False
+
+    return True
+
+
+def generate_password(pwd_len=16):
+    # List of characters to generate password from.
+    # We want to avoid to use similarly looking pairs like
+    # (O, 0), (1, l), etc.
+    lower_case_letters = 'abcdefghijkmnpqrstuvwxyz'
+    upper_case_letters = 'ABCDEFGHIJKLMPQRSTUVWXYZ'
+    digits = '23456789'
+
+    # and for punctuation we want to exclude some characters
+    # like inverted comma which can be hard to find and/or
+    # type
+    punctuation = '!#$%&()*+,-./:;<=>?@[\\]^_{|}~'
+
+    # we also want to have more letters and digits in
+    # generated password than punctuations
+    password_chars =\
+        lower_case_letters + lower_case_letters +\
+        upper_case_letters + upper_case_letters +\
+        digits + digits +\
+        punctuation
+    pwd = ''.join(random.choice(password_chars) for i in range(pwd_len))
+    logger.info('Generated password: "{}"'.format(pwd))
+    return pwd
+
+
+def store_passwords_file(pwdfile, credentials):
+    with open(pwdfile, 'wt') as f:
+        for username, pwd in credentials.items():
+            f.write('{} {}\n'.format(username, pwd))
+    return True
+
+
+def add_user(args, credentials):
+    username = args.username.lower()
+    logger.info('Adding username "{}"'.format(username))
+
+    if username in credentials:
+        logger.error('Can\'t add username {} as it is already present '
+                     'in passwords file. Do you want to run '
+                     '"update" command instead?'.format(username))
+        return False
+
+    password = args.password
+    logger.info('Adding username "{}" with password "{}"...'.format(
+                (username, password)))
+    credentials[username] = hash_password(username, password)
+
+    return store_passwords_file(args.pwdfile, credentials)
+
+
+def update_user(args, credentials):
+    username = args.username.lower()
+    logger.info('Updating username "{}"'.format(username))
+
+    if username not in credentials:
+        logger.error('Username "{}" not found in passwords file. '
+                     'Do you want to run "add" command instead?'.
+                     format(username))
+        return False
+
+    password = args.password
+    logger.info('Updating username "{}" password  to "{}"...'.format(
+                (username, password)))
+    credentials[username] = hash_password(username, password)
+    return store_passwords_file(args.pwdfile, credentials)
+
+
+def process_command(args, credentials):
+    if args.command == 'add':
+        return add_user(args, credentials)
+    elif args.command == 'update':
+        return update_user(args, credentials)
+    else:
+        logger.error('Uknown command "%s"' % args.command)

```

Comment:
Probably best to just use string formatting at this point instead of Py2 syntax

Reply:
Ah, right. Somehow I keep forgetting about the recommended way for string formatting. Thanks for catching these!

Reply:
Yep. By the way since you are on 3.6 you could simplify even further with formatted string literals, so just:

```python
f'Unknown command {args.command}'
```

Don't need to do it here but there's an open issue about it

Reply:
Great, will try to remember that.